### PR TITLE
Try to log errors that hit the global exception handler

### PR DESCRIPTION
**Pull request by mbabker**
#### Summary of Changes

This will try to write an error message about the uncaught Throwable triggering the error page to a logger registered for the `error` category while taking measures to keep the act of logging from triggering a fatal error.
#### Testing Instructions

Set up a logger for the `error` category (the debug plugin should help with this), trigger the error page, and ensure the error gets logged.

### CODE REVIEW

**Comment by mbabker** [comment]
Could use some tests here, having this would make debugging issues a little easier as anything triggering the error page would write a log statement with a stack trace if a logger is configured...

**Comment by brianteeman** [comment]
Happy to test this @mbabker but I need a little more guidance<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10341">issues.joomla.org/joomla-cms/10341</a>.</sub>

**Comment by mbabker** [comment]
Turn on debug mode and make sure the plugin is logging everything and trigger the error page.  The data from the Exception should be logged under the "error" category.

**Comment by brianteeman** [comment]
The error page in the front-end or admin?

**Comment by mbabker** [comment]
It's not dependent on either application.  All error page handling goes through this method.

**Comment by wilsonge** [comment]
Can you rebase this and I'll give it a test

**Comment by mbabker** [comment]
Done

**Comment by mbabker** [comment]
... no?

**Comment by andrepereiradasilva** [comment]
will test tonight

**Comment by zero-24** [comment]
@mbabker can you confirm that this is a correct error:

```
2016-10-03T15:16:05+00:00	CRITICAL 37.201.171.40	error	Uncaught Throwable of type JException thrown. Stack trace: #0 JRoot/libraries/legacy/error/error.php(277): JError::raise(1, 404, 'Category not fo...', NULL, true)
#1 JRoot/libraries/legacy/view/category.php(125): JError::raiseError(404, 'Category not fo...')
#2 JRoot/components/com_content/views/category/view.html.php(72): JViewCategory->commonCategoryDisplay()
#3 JRoot/libraries/legacy/controller/legacy.php(664): ContentViewCategory->display()
#4 JRoot/components/com_content/controller.php(113): JControllerLegacy->display(true, Array)
#5 JRoot/libraries/legacy/controller/legacy.php(702): ContentController->display()
#6 JRoot/components/com_content/content.php(38): JControllerLegacy->execute('')
#7 JRoot/libraries/cms/component/helper.php(405): require_once('/www/htdocs/w00...')
#8 JRoot/libraries/cms/component/helper.php(380): JComponentHelper::executeComponent('/www/htdocs/w00...')
#9 JRoot/libraries/cms/application/site.php(191): JComponentHelper::renderComponent('com_content')
#10 JRoot/libraries/cms/application/site.php(230): JApplicationSite->dispatch()
#11 JRoot/libraries/cms/application/cms.php(261): JApplicationSite->doExecute()
#12 JRoot/index.php(49): JApplicationCms->execute()
#13 {main}
```

If i produce a 404?

**Comment by mbabker** [comment]
Looks right to me; it's logging that an exception went uncaught to the global handler and the stack trace of the exception with priority "CRITICAL".

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on af2646d06525937a0fa7c24abc408ad72f733e03<br /><br />:+1: than :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10341">issues.joomla.org/joomla-cms/10341</a>.</sub>

File: libraries/cms/error/page.php [hunk 1]
```diff
@@ -35,6 +35,29 @@ public static function render($error)
...
+						),
+						JLog::CRITICAL,
+						'error'
+					);
+				}
+				catch (Throwable $e)
+				{
+					// Logging failed, don't make a stink about it though
+				}
+				catch (Exception $e)
```
Comment (alikon) [inline libraries/cms/error/page.php]: shouldn't the  2nd `catch {}` go to one tab less ?

Comment (mbabker) [inline libraries/cms/error/page.php]: Negative.  It's a trick used for PHP 5 & 7 compatibility to catch all error types.  You can have multiple catch blocks for a single try.

**Comment by alikon** [comment]
i've tested  maybe in the same way as @zero-24 
i've  visited an non-existent url an checked that no error was logged on everything.php
i've applyed the PR
again visited the same non-existent url and now in the everything.php log file the error is logged
errorlogged

File: libraries/cms/error/page.php (hunk 1 above)
Comment (alikon) [inline libraries/cms/error/page.php]: thanks for the explanation

**Comment by zero-24** [comment]
Yes it is the sameway as i have tested it @alikon

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on af2646d06525937a0fa7c24abc408ad72f733e03<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10341">issues.joomla.org/joomla-cms/10341</a>.</sub>

**Comment by zero-24** [comment]
RTC cc: @rdeutz 

Thanks 👍 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10341">issues.joomla.org/joomla-cms/10341</a>.</sub>

**Comment by brianteeman** [comment]
Removed RTC label

**Pull request merged**